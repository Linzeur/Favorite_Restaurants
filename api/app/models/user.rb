class User < ApplicationRecord

  validates :name, presence: true
  validates :email, format: { with: /[a-zA-Z0-9]*@[a-zA-Z]*.[a-zA-Z0-9]*/, message: "Bad format"  },
                    uniqueness: true

  has_secure_password
  has_secure_token
  has_secure_token :reset_digest

  has_many :locations
  has_many :favorite_restaurants

  def self.valid_login?(email, password)
    user = find_by(email: email)
    user if user&.authenticate(password)
  end

  def invalidate_token
    update(token: nil)
  end
end
