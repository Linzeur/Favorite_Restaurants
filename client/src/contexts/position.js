import React from "react";

const PositionContext = React.createContext();

function PositionProvider(props) {
  const [position, setPosition] = React.useState([0, 0, false]);

  React.useEffect(() => {
    const watchId = navigator.geolocation.watchPosition(({ coords }) => {
      setPosition([coords.latitude, coords.longitude, true]);
      sessionStorage.setItem(
        "lastCurrentPosition",
        JSON.stringify({ lat: coords.latitude, lng: coords.longitude })
      );
    });
    return () => navigator.geolocation.clearWatch(watchId);
  }, [setPosition]);

  const value = {
    latitude: position[0],
    longitude: position[1],
    callAPI: position[2],
    onChange: (lat, lng, callAPI) => {
      setPosition([lat, lng, callAPI]);
    }
  };

  return <PositionContext.Provider value={value} {...props} />;
}

export { PositionProvider, PositionContext };
