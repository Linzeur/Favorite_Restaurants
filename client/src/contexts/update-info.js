import React from "react";

const UpdateListContext = React.createContext();

function UpdateListProvider(props) {
  const [updateList, setUpdateList] = React.useState({
    locations: 0,
    favorites: 0
  });

  const value = {
    locations: updateList.locations,
    favorites: updateList.favorites,
    onChange: key => {
      setUpdateList(prevState => {
        return { ...prevState, [key]: prevState[key] + 1 };
      });
    }
  };

  return <UpdateListContext.Provider value={value} {...props} />;
}

export { UpdateListProvider, UpdateListContext };
