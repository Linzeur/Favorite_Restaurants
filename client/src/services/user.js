import { apiUrl } from "../util";

const API_USERS = `${apiUrl}users`;

async function createUser(userData) {
  const response = await fetch(API_USERS, {
    method: "POST",
    credentials: "include",
    headers: {
      "Content-Type": "application/json"
    },
    body: JSON.stringify(userData)
  });

  if (!response.ok) {
    const { errors } = await response.json();
    throw new Error(errors.message);
  }

  return response.json();
}

export { createUser };
