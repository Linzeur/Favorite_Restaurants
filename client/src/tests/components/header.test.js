import React from "react";
import { render, getNodeText } from "@testing-library/react";

import Header from "../../components/header";
import { UserProvider } from "../../contexts/user";

const user = {
  name: "User Test",
  email: "test@test.com"
};

test("header component", () => {
  localStorage.setItem("user", JSON.stringify(user));

  const { asFragment } = render(
    <UserProvider>
      <Header title={"Titulo nuevo"} />
    </UserProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test("Valid username and title shown ", () => {
  localStorage.setItem("user", JSON.stringify(user));

  const { getByLabelText } = render(
    <UserProvider>
      <Header title={"Titulo nuevo"} />
    </UserProvider>
  );

  const spanTitle = getByLabelText("Title of header for this page");
  const spanUserName = getByLabelText("Username");

  expect(getNodeText(spanTitle)).toBe("Titulo nuevo");
  expect(getNodeText(spanUserName)).toBe("User Test");
});
