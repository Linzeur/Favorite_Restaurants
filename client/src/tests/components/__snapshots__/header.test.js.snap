// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component 1`] = `
<DocumentFragment>
  <div
    class="css-ly7cin-Header"
  >
    <span
      aria-label="Title of header for this page"
      class="css-1f27ziu-Header"
    >
      Titulo nuevo
    </span>
    <div
      class="css-1jw2xof-Header"
    >
      <div
        class="css-g39h9j-Header"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
          />
        </svg>
        <span
          aria-label="Username"
          class="css-16xfdcu-Header"
        >
          User Test
        </span>
      </div>
      <div
        aria-label="log out session"
        class="css-1c4fmxo-LogOut"
        role="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M497 273L329 441c-15 15-41 4.5-41-17v-96H152c-13.3 0-24-10.7-24-24v-96c0-13.3 10.7-24 24-24h136V88c0-21.4 25.9-32 41-17l168 168c9.3 9.4 9.3 24.6 0 34zM192 436v-40c0-6.6-5.4-12-12-12H96c-17.7 0-32-14.3-32-32V160c0-17.7 14.3-32 32-32h84c6.6 0 12-5.4 12-12V76c0-6.6-5.4-12-12-12H96c-53 0-96 43-96 96v192c0 53 43 96 96 96h84c6.6 0 12-5.4 12-12z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
